components/validator/challenger.go

```diff
@@ -110,22 +111,22 @@ func (c *Challenger) GetInvalidOutputRange() (*OutputRange, error) {
 	if err != nil {
 		return nil, err
 	}
-	if nextOutputIndex.Cmp(common.Big0) == 0 {
-		c.log.Info("the output has not been submitted yet.")
+	latestOutputIndex := new(big.Int).Sub(nextOutputIndex, common.Big1)
+	if latestOutputIndex.Cmp(common.Big1) == -1 {
+		c.log.Info("challenge for genesis output is not allowed", "latestOutputIndex", latestOutputIndex)
 		return nil, nil
 	}
-	latestOutputIndex := new(big.Int).Sub(nextOutputIndex, common.Big1)
 
 	if c.checkpoint == nil {
-		waitingOutputs := new(big.Int).Div(c.finalizationPeriodSeconds, c.submissionInterval)
-		if latestOutputIndex.Cmp(waitingOutputs) == -1 {
-			// TODO(chokobole): Enable dispute resolution including genesis output root.
+		submissionIntervalSeconds := new(big.Int).Mul(c.submissionInterval, c.l2BlockTime)
+		waitingOutputNum := new(big.Int).Div(c.finalizationPeriodSeconds, submissionIntervalSeconds)
+		if latestOutputIndex.Cmp(waitingOutputNum) != 1 {
 			c.checkpoint = common.Big1
-		} else if waitingOutputs.Cmp(common.Big0) == 0 {
+		} else if waitingOutputNum.Cmp(common.Big0) == 0 {
 			c.checkpoint = latestOutputIndex
 			return nil, nil
 		} else {
-			c.checkpoint = new(big.Int).Mod(latestOutputIndex, waitingOutputs)
+			c.checkpoint = new(big.Int).Sub(nextOutputIndex, waitingOutputNum)
```

Comment:
It would be good to note this change: `Mod` to `Sub`

AFAIK, when `c.checkpoint` is nil and the `latestOutputIndex` is larger than `waitingOutputNum` (meaning that after the first finalization), the `c.checkpoint` should be the start point of the finalization window.